core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -270,6 +272,14 @@ class ClientQuotaManager(private val config: ClientQuotaManagerConfig,
     }
   }
 
+  def logQuotaViolations(): Unit = {
+    quotaViolationStatBySensorId.forEach {
```

Comment:
I do not know the details on how ConcurrentHashMap manages concurrency. But in general, if it uses lock mechanism, this method would block other threads for a significant amount of time if the map has a lot of items.